test/deployments.ts

```diff
@@ -134,6 +134,89 @@ describe("Aludel factory deployments", function () {
           expect(program.name).to.eq("some name");
         });
+        describe("WHEN updates the program using update-program", () => {
+          let program: AludelFactory.ProgramDataStruct;
+          beforeEach(async () => {
+            program = await factory.programs(preexistingProgram.address);
+          });
+          it("AND updates name", async () => {
+            beforeEach(async () => {
+              await run("update-program", {
+                program: preexistingProgram.address,
+                newName: "a brave new name",
+                newUrl: "https://new.domain",
+              });
+            });
+            it("THEN only the name is changed", async () => {
+              const updatedProgram = await factory.programs(
+                preexistingProgram.address
+              );
+              expect(updatedProgram.name).to.eq("a brave new name");
+              expect(updatedProgram.stakingTokenUrl).to.eq(
+                program.stakingTokenUrl
+              );
+              expect(updatedProgram.template).to.eq(program.template);
+              expect(updatedProgram.startTime).to.eq(program.startTime);
+            });
+          });
+          describe("AND updates url", async () => {
+            beforeEach(async () => {
+              await run("update-program", {
+                program: preexistingProgram.address,
+                newUrl: "https://new.domain",
+              });
+            });
+            it("THEN only the url is changed", async () => {
+              const updatedProgram = await factory.programs(
+                preexistingProgram.address
+              );
+              expect(updatedProgram.name).to.eq(program.name);
+              expect(updatedProgram.stakingTokenUrl).to.eq(
+                "https://new.domain"
+              );
+              expect(updatedProgram.template).to.eq(program.template);
+              expect(updatedProgram.startTime).to.eq(program.startTime);
+            });
+          });
+          describe("AND update name and url", async () => {
+            beforeEach(async () => {
+              await run("update-program", {
+                program: preexistingProgram.address,
+                newName: "a brave new name",
+                newUrl: "https://new.domain",
+              });
+            });
+            it("THEN only the name and the url are changed", async () => {
+              const updatedProgram = await factory.programs(
+                preexistingProgram.address
+              );
+              expect(updatedProgram.name).to.eq("a brave new name");
+              expect(updatedProgram.stakingTokenUrl).to.eq(
+                "https://new.domain"
+              );
+              expect(updatedProgram.template).to.eq(program.template);
+              expect(updatedProgram.startTime).to.eq(program.startTime);
+            });
+          });
+          describe("AND updates nothing", async () => {
```

Comment:
I believe the task should error out in this case, there's no need for it to do nothing and it'd make the tests simpler. (and would be a bit more consistent with the behaviour of `update-template`, which errors out if none of the optional flags are passed